test/selenium/README.md

```diff
@@ -22,18 +22,106 @@ Run the [run-tests script](https://github.com/cockpit-project/cockpit/blob/maste
+    ```
+    ssh -L $YOUR_IP:9991:127.0.0.2:9091 $(whoami)@localhost
+    ```
+  - Run the test. ``HUB_IP`` is ip address of your selenium grid, has to be able to see cockpit instance via port 9991 on ``GUEST_IP``
+    ```
+    test/selenium/run-tests --hub $HUB_IP:$GUEST_IP:9991 -v -b chrome selenium-base.py --sit
+    ```
+
+#### Selenium grid on your machine with your local browsers
+Steps are described in [selenium page](https://www.browserstack.com/guide/selenium-grid-tutorial)
+  - Download selenium standalone:
+    ```
+    curl -f -L https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar > selenium.jar
+    ```
+  - Run hub and attach browsers here:
+    ```
+    java -jar selenium.jar -role hub
+    ```
+    - Then attach your local browsers there. You have browsers eg. firefox or chrome installed, and also ensure you have proper [selenium drivers for selected the browser](https://github.com/cockpit-project/cockpit/tree/master/test/selenium#which-browser-to-use)
+      ```
+      java -jar selenium.jar -role node -hub https://localhost:4444/grid/register
+      ```
+  - Run the test. ``HUB_IP`` is ip address of your selenium grid
+    ```
+    test/selenium/run-tests --hub $HUB_IP -v -b chrome selenium-base.py --sit
+    ```
+#### Reschedule the the test/s again
+It is possible to reschedule tests if you used ``--sit`` option for ``run-tests``, in case of failure, machines won't be destroyed.
+Command-line options depend on which selenium grid you used.
+It is described on the output of the ``run-tests`` script. Eg.:
+
+```
+$test/selenium/run-tests --hub $HUB_IP -v -b chrome selenium-navigate.py --sit
+
+...
+To rerun tests connect to VM machine with avocado installed via
+    ssh -p 2202 root@127.0.0.2
+
+and run the command
+    PYTHONPATH=/tmp/avocado_library HUB=$HUB_IP_ADDR GUEST=127.0.0.2
+PORT=9091 SSH_PORT=22 SSH_GUEST=10.111.113.1 BROWSER=chrome timeout 297
+python3 -m avocado run --show-job-log /tmp/avocado_tests/selenium-navigate.py:NavigateTestSuite.testNavigateNoReload 2>&1
+```
+
+It is possible also to run the tests on your machine, without test VM (bots/images) what contains installed avocado testing framework.
+  - You have to install python3 ``avocado`` via rpm or pip
+  - Then a little bit modify command what you see on the output of the ``run-tests`` command and it depends how you run the selenium grid.
+    ```
+    HUB=$HUB_IP GUEST=$GUEST_IP PORT=9991 SSH_PORT=2021 SSH_GUEST=127.0.0.2 BROWSER=chrome python3 -m avocado --show=test run  test/selenium/selenium-navigate.py
+    ```
+   - **Remember**
+     - ``GUEST`` has to be accessible from ``HUB`` via ``PORT`` number
+     - Your avocado runner has to be able to connect ``SSH_GUEST`` via ``SSH_PORT`` (what is same machine as ``GUEST`` but may have another address, because of port forwarding and ``HUB`` location)
```

Comment:
to connect to ``SSH_GUEST``

Reply:
We tend to say connect to Server on Port, but via port. But nitpick here.

Reply:
I don't really get the sentence in the parenthesis.
(what is same... 
I guess you want to say `which is the same machine`.. But the avocado runner is the same machines as the machine specified by GUEST or what?

Reply:
> I don't really get the sentence in the parenthesis.
> (what is same...
> I guess you want to say `which is the same machine`.. But the avocado runner is the same machines as the machine specified by GUEST or what?

no no, avocado runner is currently separate VM inside run-tests script. I wrote it is this generic way, to show, that it is not important where avocado runner is VM inside run-tests, or if it  is your machine, or whatever. I understand that this could be confusing. I'll try to rewrite it somehow